pop-cli-for-appchains/guides/running-on-paseo-locally.md

```diff
@@ -417,51 +412,117 @@ curl -H "Content-Type: application/json" \
 
-Use sudo privileges on the Relay chain to force a lease for our parachain:
+```bash
+pop call chain --url ws://localhost:57731
+```
 
-<figure><img src="../.gitbook/assets/Screenshot 2024-09-24 at 11.46.09 AM.png" alt=""><figcaption><p>Force a lease</p></figcaption></figure>
+ 
+```bash
+┌   Pop CLI : Call a chain
+│
+◇  What would you like to do?
+│  Reserve a parachain ID 
+│
+◇  Signer of the extrinsic:
+│  <STASH ACCOUNT>
+│  
+...
+       Event Balances ➜ Reserved
+         who: <STASH ACCOUNT>
+         amount: 100UNIT
+       Event Registrar ➜ Reserved
+         para_id: Id(2000)
+         who: <STASH ACCOUNT>
+...         
+```
 
-> In production, leases are assigned via the result of an on-chain auction.
+In the events we can see the `para_id` that is assigned to the chain. Make sure this is the para ID specified in the chain spec file (and thus the chain artifacts).
 
-Wait for the lease to take effect:
+### Register Para ID with Genesis State & Code
 
-<figure><img src="../.gitbook/assets/Screenshot 2024-09-24 at 11.46.30 AM.png" alt=""><figcaption><p>Upgrading Parathread</p></figcaption></figure>
+Now we register the para ID with the generated genesis state (`para-2000-genesis-state`) and genesis code (`para-2000.wasm`).
 
-You will then see the parathread upgraded to a parachain:
+```bash
+│
+◇  Do you want to perform another call?
+│  Yes
+│
+◇  What would you like to do?
+│  Register a parachain ID with genesis state and code
+│
+◇  Enter the value for the parameter: id
+│  2000
+│
+◇  The value for `genesis_head` might be too large to enter. You may enter the path to a file instead.
+│  para-2000-genesis-state
+│
+◇  The value for `validation_code` might be too large to enter. You may enter the path to a file instead.
+│  para-2000.wasm
+│
+◇  Signer of the extrinsic:
+│  <STASH ACCOUNT>
+│
+...
+       Event Balances ➜ Withdraw
+         who: <STASH ACCOUNT>
+         amount: 90.71989507390UNIT
+       Event Balances ➜ Reserved
+         who: <STASH ACCOUNT>
+         amount: 3.145826kUNIT
+       Event Paras ➜ PvfCheckStarted
+         0: ValidationCodeHash(0x1821617486094e18595084b580fe9324a084adedbf80ec61c9d7b75736ab5f5b)
+         1: Id(2000)
+       Event Registrar ➜ Registered
+         para_id: Id(2000)
+         manager: <STASH ACCOUNT>
+...
+```
 
-<figure><img src="../.gitbook/assets/Screenshot 2024-09-24 at 11.50.18 AM.png" alt=""><figcaption><p>Parachain Tab</p></figcaption></figure>
+Your chain is now registered on Paseo!
 
-After a few moments, you should see the parachain producing blocks. The included / backed will update with the Relay chain's block number in which your parachain's block was included and backed. You can also check your collator logs to confirm.
+### Buy an On Demand Core
 
-<figure><img src="../.gitbook/assets/Screenshot 2024-09-24 at 2.21.37 PM.png" alt=""><figcaption><p>Notice the included / backed</p></figcaption></figure>
+Now we need to buy a core to have Paseo validate a block.
```

Comment:
Not 100% correct. One only buy the finalization of one core.
More of a pay finalization as you produce blocks.

Buying a core would be something more akin to buying bulk coretime. Which in reality isn't cores either what is being sold, but regions. :)

Reply:
validated and finalised*